appsec-kit-v7/src/main/java/com/vaadin/appsec/v7/service/NotificationInitializer.java

```diff
@@ -59,33 +65,43 @@ private void subscribeSessionToScanEvents(SessionInitEvent e) {
         session.addUIProvider(new AppSecUIProvider());
         AppSecService appSecService = AppSecService.getInstance();
         Registration scanEventRegistration = appSecService
-                .addScanEventListener(scanEvent -> {
-                    int newVulns = scanEvent.getNewVulnerabilities().size();
+                .addScanEventListener(event -> {
+                    int newVulns = event.getNewVulnerabilities().size();
                     if (isSessionOpen(session) && newVulns > 0) {
-                        session.getUIs().forEach(this::doNotifyUI);
+                        Collection<UI> uis = session.getUIs();
+                        LOGGER.debug("Notifying {} UIs for session {}",
+                                uis.size(), session.getSession().getId());
+                        session.access(
+                                () -> uis.forEach(ui -> doNotifyUI(ui, event)));
                     }
                 });
         scanEventRegistrations.put(session, scanEventRegistration);
     }
 
-    private void doNotifyUI(UI ui) {
+    private void doNotifyUI(UI ui, AppSecScanEvent event) {
         if (ui instanceof AppSecUI) {
             return;
         }
-        String link = "<a href=\"?"
+        List<Vulnerability> vulns = event.getNewVulnerabilities();
+        Map<SeverityLevel, Long> countBySeverity = vulns.stream()
+                .collect(Collectors.groupingBy(Vulnerability::getSeverityLevel,
+                        Collectors.counting()));
+        String link = "<a class=\"appsec-notification-button\" href=\"?"
                 + AppSecUIProvider.VAADIN_APPSEC_KIT_URL_PARAM
-                + "\" target=\"_blank\">here</a>";
-        String msg = "New vulnerabilities found! Click " + link
-                + " to open Vaadin AppSec Kit,"
-                + " or click on this message to dismiss it.";
-        Notification n = new Notification("Vaadin AppSec Kit", msg,
+                + "\" target=\"_blank\">Open AppSec Kit</a>";
```

Comment:
Should we also remove the `Vaadin` from page [title](https://github.com/vaadin/appsec-kit/blob/fix/notification-access/appsec-kit-v8/src/main/java/com/vaadin/appsec/v8/ui/AppSecUI.java#L32) and app [title](https://github.com/vaadin/appsec-kit/blob/fix/notification-access/appsec-kit-v8/src/main/java/com/vaadin/appsec/v8/ui/content/MainView.java#L55)? And also from V7.

Reply:
Sure, removed also there.